impeller/entity/contents/text_contents.cc

```diff
@@ -25,6 +25,11 @@ TextContents::TextContents() = default;
 
 TextContents::~TextContents() = default;
 
+// static
+Scalar TextContents::RoundFontScale(Scalar value) {
+  return std::round(value * 10.0) / 10.0;
```

Comment:
I think this snapping would need to be done logarithmically (apply log, apply snapping, invert with exponential)? Also, the snapping precision would need to somehow incorporate the text frame's font size as well.

For example, the difference between 0.1 and 0.2 is potentially huge, getting worse as the given font size increases.

Reply:
hmm thats a good point. I'll need to investigate this a bit more.

Reply:
okay, after much thinking I've come up with an OK solution.... round to one decimal if the point size is < 25, Round to two decimals if > 50.  🤷‍♂️

Reply:
I don't think this covers all cases, would something like this work?

```cpp
// The point_size is multiplied into the total scale of the font later,
// and so we need to linearly increase the step precision as the point size increases.
auto precision = 20 * point_size; // The constant here may need to be adjusted for best perf.
return std::pow(kE, std::round(std::log(scale) * precision) / precision);
```

![image](https://github.com/flutter/engine/assets/919017/1a9f7028-2b9d-41c4-9a8e-afa2ee09915f)

https://www.desmos.com/calculator/sqpcws9bfv